src/site/content/en/blog/responsiveness/index.md

```diff
@@ -0,0 +1,302 @@
+
+#### Average interaction latency over budget
+
+The total over-budget interaction latency divided by the total number of
+interactions. Using the above example, the score would be `(20 + 0 + 10 + 30) /
+4 = 15 ms`.
+
+#### High quantile approximation
+
+As an alternative to computing the largest interaction latency over budget, we
+also considered using a high quantile approximation, which should be fairer to
+web pages that have a lot of interactions and may be more likely to have large
+outliers. We've identified two potential high-quantile approximation strategies
+we like:
+
+- **Option 1:** keep track of the largest interaction over budget. After every
+  50 interactions, discard the largest and replace it with the next largest.
+- **Option 2:** compute the largest 10 interactions over budget and choose a
+  value from that list depending on the total number of interactions. Given N
+  total interactions, select the (N / 50 + 1)th largest value, or the 10th value
+  for pages with more than 500 interactions.
+
+## Measure these options in JavaScript
+
+The following code example can be used to determine the values of the first
+three strategies presented above. Note that it's not yet possible to measure the
+total number of interactions on a page in JavaScript, so this example doesn't
+include the average interaction over budget strategy or the high
+quantile approximation strategies.
+
+```js
+const interactionMap = new Map();
+
+let worstLatency = 0;
+let worstLatencyOverBudget = 0;
+let totalLatencyOverBudget = 0;
+
+new PerformanceObserver((entries) => {
+  for (const entry of entries.getEntries()) {
+    // Ignore entries without an interaction ID.
+    if (entry.interactionId > 0) {
+      // Get the interaction for this entry, or create one if it doesn't exist.
+      let interaction = interactionMap.get(entry.interactionId);
+      if (!interaction) {
+        interaction = {entries: []};
+        interactionMap.set(entry.interactionId, interaction);
+      }
+      interaction.entries.push(entry);
+
+      const latency = Math.max(entry.duration, interaction.latency || 0);
+      worstLatency = Math.max(worstLatency, latency);
+
+      const budget = entry.name.includes('key') ? 50 : 100;
+      const latencyOverBudget = latency - budget;
+      worstLatencyOverBudget =
+          Math.max(latencyOverBudget, worstLatencyOverBudget);
+
+      // If this event adds additional latency, update the total over budget.
+      const newLatency = latency - (interaction.latency || 0);
+      if (newLatency > 0) {
+        totalLatencyOverBudget += newLatency;
+      }
+
+      // Set the latency on the interaction so future events can reference.
+      interaction.latency = latency;
+
+      // Log the updated metric values.
+      console.log({
+        worstLatency,
+        worstLatencyOverBudget,
+        totalLatencyOverBudget,
+      });
+    }
+  }
+}).observe({type: 'event', buffered: true, durationThreshold: 50});
```

Comment:
Is it worth pointing out what these options are/linking to MDN for them?

Reply:
Probably...unfortunately the Event Timing API reference on MDN doesn't include this option (it's pretty new). I'll add a comment.